### DISCUSSION TITLE
Feature Request: Add `options` prop to `VsSelect` component

### DISCUSSION THREAD

**Discussion by mathiasayivor** [vuesax-alpha/vuesax-alpha#56] [Ideas]
### PROBLEM

Currently, when working with the `VsSelect` component, options can be added using `<VsOption>`. While this approach gets the job done, it becomes quite cumbersome in larger projects. The need to repeat this code wherever `VsSelect` is used leads to increased verbosity and can make the codebase harder to maintain.
E.g:
```vue
<vs-select v-bind="otherProps">
  <vs-option v-for="option in options" v-bind="option" />
  <!-- maybe other options here? -->
</vs-select>
```
One way to address this is for users to make their own custom component that wraps `VsSelect` (similar to the example above). But what if users are working on several different projects?

### PROPOSED SOLUTION

I propose adding a new prop: `options`, to the `VsSelect` component. This prop would automate the creation of `VsOptions`, simplifying the code and enhancing its maintainability.
E.g: Instead of writing:
```vue
<vs-select v-bind="otherProps">
  <vs-option v-for="option in options" v-bind="option" />
  <!-- maybe other options here? -->
</vs-select>
```
The user would instead write:
```vue
<vs-select v-bind="otherProps" :options="options" />
```

### ADDITIONAL CONSIDERATIONS

To ensure flexibility, I suggest offering various slots for customising both the `VsSelect` and `VsOption` components. This would enable users to tailor the components to their specific needs:
E.g:
```vue
<vs-select v-bind="otherProps" :options="options">
  <template #vs-option="vsOptionProps">
    <vs-option v-bind="vsOptionProps">
      <!-- Other customisations here -->
    </vs-option>
  </template>
  <!-- other slots here -->
</vs-select>
```

### EVEN BETTER!

To make `VsSelect` even more versatile, the following additional props can be introduced:

`option-label` [string] and `option-value` [string] props: These props would allow users to specify how to extract the `label` and `value` from objects in the `options` array. This is especially useful when dealing with objects from different sources that may have different property names (e.g. data from a backend)

`emit-value` [boolean] prop: This prop would provide users with the option to decide whether `VsSelect` should emit the selected object or just the `value`.

Array of strings as options: Users could provide an array of strings to the `options` prop, where the string would serve as both the label and the value.

Example 1 (without `emit-value` prop):
```vue
<script setup lang="ts">
interface User {
  id: number;
  displayName: string;
}

const users = ref<User[]>([
  { id: 1, displayName: 'Kelly' },
  { id: 2, displayName: 'Kelvin' },
]);

const selectedUsers = ref<User[]>([]); // An array of objects each representing a selected user
</script>
<template>
  <vs-select
    v-model="selectedUsers"
    :options="users"
    option-label="displayName"
    option-value="id"
  />
</template>
```

Example 2 (with `emit-value` prop):
```vue
<script setup lang="ts">
interface User {
  id: number;
  displayName: string;
}

const users = ref<User[]>([
  { id: 1, displayName: 'Kelly' },
  { id: 2, displayName: 'Kelvin' },
]);

const selectedUsers = ref<number[]>([]); // An array of numbers each representing the id of the selected user
</script>
<template>
  <vs-select
    v-model="selectedUsers"
    :options="users"
    option-label="displayName"
    option-value="id"
    emit-value
  />
</template>
```

Example 3 (array of strings as `options`):
```vue
<script setup lang="ts">
const fruits = ref(['Orange', 'Apple', 'Banana']);

const selectedFruits = ref<string[]>([]); // An array of strings each representing the selected fruit
</script>
<template>
  <vs-select v-model="selectedFruits" :options="fruits" />
</template>
```

**Comment by tranthinh-coding** [comment]
I tried this before, however my logic had some performance issues so I scrapped it. I may consider this case in the near future.

**Comment by mathiasayivor** [comment]
> I tried this before, however my logic had some performance issues so I scrapped it. I may consider this case in the near future.

Sure! But, the current design becomes unmaintainable when the project grows.
